eng/resolveContract.targets

```diff
@@ -12,8 +12,8 @@
     <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and
                                      '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and
                                      $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), '$(NETCoreAppCurrentVersion)'))">$(NetCoreAppCurrentRefPath)$(TargetFileName)</ContractAssemblyPath>
-    <ContractAssemblyPath Condition="'$(ContractAssemblyPath)' == '' and !$(TargetFramework.Contains('-'))">$([MSBuild]::NormalizePath('$(BaseOutputPath)', 'ref', '$(TargetFramework)-$(Configuration)', '$(TargetFileName)'))</ContractAssemblyPath>
-
+    <TargetFrameworkWithoutPlatform>$([System.Text.RegularExpressions.Regex]::Replace('$(TargetFramework)', '(-[^;]+)', ''))</TargetFrameworkWithoutPlatform>
```

Comment:
Internal property `_`? In retrospective maybe we should have really added that to the sdk as we are now doing the same thing in multiple places... :(

Reply:
yeah :P, i will change to internal here for this pr